pkg/uefivars/guid.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2015-2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+package uefivars
+
+import (
+	"github.com/google/uuid"
+)
+
+//A mixed-endianness guid, as used by MS and UEFI. For normal uuid-related functionality, use a  package like github.com/google/uuid .
+type MixedGuid [16]byte
+
+//Converts MixedGuid to a uuid.UUID
+func (m MixedGuid) ToStdEnc() (u uuid.UUID) {
+	u[0], u[1], u[2], u[3] = m[3], m[2], m[1], m[0]
+	u[4], u[5] = m[5], m[4]
+	u[6], u[7] = m[7], m[6]
+	copy(u[8:], m[8:])
+	return
+}
+
+//Converts uuid.UUID to MixedGuid
+func FromStdEnc(u uuid.UUID) (m MixedGuid) {
```

Comment:
the only reason that google/uuid is a dependency here is because of this func -- could we remove that? for our other use cases, this would just add some binary size

Reply:
I'd not used google/uuid originally, not sure why the dep crept in. Will fix.

Reply:
fixed, though now I wonder if the guid/uuid functionality should be its own package.